src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -624,7 +636,6 @@ private function walkGroupAst(TreeNode $ast, TrinaryLogic &$isNonEmpty, TrinaryL
 					}
 
 					if (Strings::match($literalValue, '/^\d+$/') === null) {
-						$allNumeric = false;
```

Comment:
I played a bit more with it. I have a feeling this whole `if ($ast->getId() === '#range' || $ast->getId() === '#class') {` can be removed, because we achieve the same thing already via recursive `walkGroupAst()` and handling of the `if ($ast->getId() === 'token') {` above (ranges and classes can only contain tokens, right?)

Reply:
Yeah that is possible, but can we maybe just let this sit until @ondrejmirtes can review and then bring further cleanups later? :D

Reply:
Ok I've done it after all in https://github.com/phpstan/phpstan-src/pull/3244/commits/64901f62c36a17e4e9a54243b795e2affbcadd86 as I was modifying the PR again anyway.. Looks good :)